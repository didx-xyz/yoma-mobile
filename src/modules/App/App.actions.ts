import { createAction } from '@reduxjs/toolkit'

import { AppActions } from './types'

export const resetAppData = createAction(AppActions.resetApp)
