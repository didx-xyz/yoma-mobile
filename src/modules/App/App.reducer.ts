import { createAction } from '@reduxjs/toolkit'

const name = '[App]'

export const resetAppData = createAction(`${name} resetAppData`)
export const hydrateApp = createAction(`${name} hydrateApp`)
