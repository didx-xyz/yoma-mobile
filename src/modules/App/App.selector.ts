import { createSelector } from '@reduxjs/toolkit'

import * as AuthSelectors from '../Auth/Auth.selector'

export default createSelector(AuthSelectors.selectIsAuthenticated, isAuthorised => ({
  isAuthorised,
}))
