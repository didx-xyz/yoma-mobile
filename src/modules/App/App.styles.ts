import { StyleSheet, TextStyle, ViewStyle } from 'react-native'
import { Colors } from 'react-native/Libraries/NewAppScreen'

const styles = {
  scrollView: {
    backgroundColor: Colors.lighter,
  } as ViewStyle,
  engine: {
    position: 'absolute',
    right: 0,
  } as ViewStyle,
  body: {
    backgroundColor: Colors.white,
  } as ViewStyle,
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  } as ViewStyle,
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  } as TextStyle,
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  } as TextStyle,
  highlight: {
    fontWeight: '700',
  } as TextStyle,
  footer: {
    color: Colors.dark,
    fontSize: 12,
    fontWeight: '600',
    padding: 4,
    paddingRight: 12,
    textAlign: 'right',
  } as TextStyle,
}
export default StyleSheet.create(styles)
