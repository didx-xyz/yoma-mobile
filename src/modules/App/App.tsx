import React, { useEffect } from 'react'
import FlashMessage from 'react-native-flash-message'

import { RNLocalize, setI18nConfig } from '~/locales/i18n'
import Sentry from '~/monitoring'

import Navigation from '../Navigation'

setI18nConfig()

const App = () => {
  useEffect(() => {
    RNLocalize.addEventListener('change', setI18nConfig)
    return () => {
      RNLocalize.removeEventListener('change', setI18nConfig)
    }
  }, [])

  return (
    <>
      <Navigation />
      <FlashMessage position="top" />
    </>
  )
}
export default Sentry.wrap(App)
