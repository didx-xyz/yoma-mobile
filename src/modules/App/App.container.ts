import { connect } from 'react-redux'

import App from './App'
import selector from './App.selector'

const mapStateToProps = selector
const mapDispatchToProps = null

export default connect(mapStateToProps, mapDispatchToProps)(App)
