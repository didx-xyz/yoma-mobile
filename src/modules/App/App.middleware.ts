import { Middleware } from 'redux'

import * as AuthActions from './../Auth/Auth.reducer'
import { resetAppData } from './App.reducer'

export const appResetFlow: Middleware = ({ dispatch }) => next => action => {
  const result = next(action)

  if (resetAppData.match(action)) {
    dispatch(AuthActions.clearAuth())
  }

  return result
}
