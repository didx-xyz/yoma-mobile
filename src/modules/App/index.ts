import App from './App'
import * as appMiddleware from './App.middleware'
import * as actions from './App.reducer'

export { actions, appMiddleware }
export default App
