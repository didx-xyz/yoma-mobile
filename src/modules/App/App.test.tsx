import { cleanup, render } from '@testing-library/react-native'
import React from 'react'
import 'react-native'

import App from './App'

jest.mock('react-native-localize', () => ({
  findBestAvailableLanguage: (_x: any) => ({
    languageTag: null,
  }),
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}))
jest.mock('@react-native-google-signin/google-signin', () => ({}))

jest.mock('rn-fetch-blob', () => ({}))

describe('App', () => {
  afterEach(cleanup)
  it('should render correctly', () => {
    const { toJSON } = render(<App />)
    const componentAsJSON = toJSON()

    expect(componentAsJSON).toMatchSnapshot()
  })
})
