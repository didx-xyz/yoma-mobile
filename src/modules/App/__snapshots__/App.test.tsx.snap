// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<View
  style={
    Object {
      "left": 0,
      "opacity": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -48,
        },
      ],
      "zIndex": 99,
    }
  }
/>
`;
